app/src/main/java/com/github/ybecker/epforuml/NewQuestionFragment.kt

```diff
@@ -101,13 +103,26 @@ class NewQuestionFragment(val mainActivity: MainActivity) : Fragment() {
 
                 // If the course is found, add the question to the database and navigate to the home screen
                 if (course != null) {
-                    DatabaseManager.db.addQuestion(
-                        user.userId,
-                        course.courseId,
-                        questTitle.text.toString(),
-                        questBody.text.toString(),
-                        imageURI.toString()
-                    )
+                    if(imageURI.text=="null"){
+                        db.addQuestion(
+                            user.userId,
+                            course.courseId,
+                            questTitle.text.toString(),
+                            questBody.text.toString(),
+                            imageURI.text.toString()
+                        ).join()
+                    }else{
+                        db.addQuestionWithUri(
+                            user.userId,
+                            course.courseId,
+                            questTitle.text.toString(),
+                            questBody.text.toString(),
+                            imageURI.text.toString(),
+                            this.mainActivity
+                        ).join()
```

Comment:
Maybe use a thenAccept() instead of the join() as the future will wait for the database. For some reason joins on firebase database methods tend to loop forever so it might be safer to remove it.